client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -842,32 +858,34 @@ public SendResult send(Message msg, MessageQueue mq, long timeout)
     /**
      * KERNEL ASYNC -------------------------------------------------------
      */
-    public void send(Message msg, MessageQueue mq, SendCallback sendCallback)
-        throws MQClientException, RemotingException, InterruptedException {
+    public void send(Message msg, MessageQueue mq, SendCallback sendCallback) {
         send(msg, mq, sendCallback, this.defaultMQProducer.getSendMsgTimeout());
     }
 
-    public void send(Message msg, MessageQueue mq, SendCallback sendCallback, long timeout)
-        throws MQClientException, RemotingException, InterruptedException {
-        this.makeSureStateOK();
-        Validators.checkMessage(msg, this.defaultMQProducer);
-
-        if (!msg.getTopic().equals(mq.getTopic())) {
-            throw new MQClientException("message's topic not equal mq's topic", null);
-        }
+    public void send(final Message msg, final MessageQueue mq, final SendCallback sendCallback, final long timeout) {
+        this.getCallbackExecutor().submit(new Runnable() {
```

Comment:
This method will put a really heavy task into PublicExecutor if the client doesn't set CallBack executor.

Reply:
sure, how do we separate and assure our executor work well?

Reply:
How about seperate the public executor and callback executor by default?